### PR TITLE
[AArch64][SVE] Add e2e tests for small and large matmuls

### PR DESCRIPTION
**Pull request by MacDue**
This requires #15276, which prevents IREE from generating scalable vector types that are poorly supported by the LLVM backend (and will likely result in it aborting).

These tests are tagged as requiring aarch64+sve, so currently won't be run in CI (as there's no SVE capable runners yet). When run locally (on a machine with SVE) these tests pass, and do use scalable vectorization.

ci-extra: build_test_all_windows,build_test_all_macos_arm64,build_test_all_macos_x86_64

### CODE REVIEW

File: tests/e2e/matmul/CMakeLists.txt [hunk 1]
```diff
@@ -29,6 +29,31 @@ iree_generated_trace_runner_test(
...
+    "llvm-cpu"
+  DRIVERS
+    "local-task"
+  COMPILER_FLAGS
+    "--iree-opt-data-tiling=false"
+  LABELS
+
+  TARGET_CPU_FEATURES_VARIANTS
+    "default"
 )
```
Comment (banach-space) [inline tests/e2e/matmul/CMakeLists.txt:57]: This change is unrelated to SVE, right?

Comment (MacDue) [reply tests/e2e/matmul/CMakeLists.txt:57]: I wanted to update the existing LLVMCPU tests to cover large/small matmuls with SVE. The widening tests are not supported, so currently SVE is not enabled there. Note this CMake file is generated from the BUILD.bazel file.

**Review by banach-space — APPROVED** [review]
This makes sense to me, thanks! Also, from what I can tell these are not included in the CI workflows so won't be run on hardware that does not support SVE. So should be safe to land :)

*EDIT* I spoke too fast: https://discord.com/channels/689900678990135345/1166024193599615006/1167517090912161904

@ScottTodd Could you take a look?

File: tests/e2e/matmul/CMakeLists.txt [hunk 2]
```diff
@@ -50,6 +75,33 @@ iree_generated_trace_runner_test(
     "--iree-opt-data-tiling=false"
   LABELS
 
+  TARGET_CPU_FEATURES_VARIANTS
+    "default"
+    "arm_64:sve:+sve"
+)
```
Comment (ScottTodd) [inline tests/e2e/matmul/CMakeLists.txt:81]: Looks like this check is working as intended to exclude tests from certain platforms: https://github.com/openxla/iree/blob/39afb04919fa5205ed074e8dceb1b727449038a8/build_tools/cmake/iree_trace_runner_test.cmake#L341-L346

Windows and macOS builds are passing.

**Review by ScottTodd — APPROVED** [review]
